src/beamlime/dashboard/job_service.py

```diff
@@ -72,23 +157,63 @@ def data_updated(self, updated_keys: set[ResultKey]) -> None:
             # output_name is None.
             source_data = all_source_data.setdefault(source_name, {})
             source_data[key.output_name] = value
-        if notify_job_update:
-            self._notify_job_update()
+        if notify_job_data_update:
+            self._notify_job_data_update()
 
-    def _notify_job_update(self) -> None:
+    def _notify_job_data_update(self) -> None:
         """Notify listeners about job updates."""
         # For now we just log the job updates. In future we may want to have a more
         # sophisticated notification mechanism.
-        self._logger.info("Job data updated for jobs: %s", list(self._job_info.keys()))
+        self._logger.debug("Job data updated for jobs: %s", list(self._job_info.keys()))
 
         # Notify all subscribers
-        for callback in self._job_update_subscribers:
+        for callback in self._job_data_update_subscribers:
             try:
                 callback()
             except Exception as e:
                 self._logger.error("Error in job update callback: %s", e)
 
+    def _notify_job_status_update(self) -> None:
+        """Notify listeners about job status updates."""
+        self._logger.debug(
+            "Job statuses updated for jobs: %s", list(self._job_statuses.keys())
+        )
+
+        # Notify all subscribers
+        for callback in self._job_status_update_subscribers:
+            try:
+                callback()
+            except Exception as e:
+                self._logger.error("Error in job status update callback: %s", e)
+
+    def remove_job(self, job_id: JobId) -> None:
```

Comment:
Does the order of steps in the function matter? If so, can you add a comment about it?

Reply:
It should not matter, since currently `JobManager` is designed for single-threaded use.